python/lsst/source/injection/inject_coadd.py

```diff
@@ -82,3 +145,175 @@ def runQuantum(self, butler_quantum_context, input_refs, output_refs):
         input_keys = ["injection_catalogs", "input_exposure", "sky_map", "psf", "photo_calib", "wcs"]
         outputs = self.run(**{key: value for (key, value) in inputs.items() if key in input_keys})
         butler_quantum_context.put(outputs, output_refs)
+
+    """
+    Establish the variance scale by a linear fit of variance vs. flux.
+    In practice we see that most pixels in a coadd obey a consistent, simple
+    linear relationship between variance and flux, but a small sample do not.
+
+    To identify and hence ignore these odd pixels, we perform two rounds of
+    RANSAC fits. RANSAC iteratively finds a least-squares straight line fit on
+    random subsamples of points, while identifying outliers. The inlier pixels
+    from a first RANSAC fit tend to be regions of empty space and the extreme
+    outer edges of galaxies, while the outliers tend to be the inner regions of
+    galaxies.
+
+    We can pick up some more pixels in the galaxies by running a second RANSCAC
+    fit on just the outliers of the first fit. In the second round, the inliers
+    tend to be the bulk of the galaxy, while the outliers are the innermost
+    cores of galaxies. In practice, the inliers from this second round tend to
+    have a qualitatively similar variance-vs-flux relationship to the outliers
+    from the first fit and do not strongly alter the final variance scale we
+    get; while the outliers from the second round are truly wild and should
+    clearly be omitted from a simple linear fit.
+
+    In both rounds, random variation of the points sampled by the RANSAC fit
+    causes the fitted slope and intercept to vary, and sometimes the fit can
+    settle on a pathological sample of points as its inliers. We try to avoid
+    such pathologies by running each fit multiple times with different seeds,
+    and using K-Means clustering on the resulting slopes to identify the most
+    stable value.
+    """
```

Comment:
This feels like it should instead go into the doc string for the `get_variance_scale` method, perhaps some (all?) under `Notes`.